src/map/data.ts

```diff
@@ -195,46 +170,39 @@ function createMapCollectionHref(collection: RawMapCollection, layers: RawMapLay
 
 function normalizeLegendItems(
   collectionId: string,
-  legendItems: LegendItem[],
+  legendItem: RawLegendItem,
   layers: RawMapLayer[],
-): MixedLegendItem[] {
-  return legendItems.map((legendItem) => {
-    const mapLayer = legendItem.id ? findBy(layers, 'id', legendItem.id) : null
-    const notSelectable = legendItem.notSelectable || !legendItem.id // Legend items with an id are always selectable, unless specified otherwise.
+): MapLayerLegendItem {
+  const mapLayer = legendItem.id ? findBy(layers, 'id', legendItem.id) : null
 
-    // Merge the legend item with the map layer if found.
-    if (legendItem.id && mapLayer) {
-      const params = mapLayer.params
-        ? querystring.stringify(mapLayer.params as ParsedUrlQueryInput)
-        : undefined
-
-      return {
-        ...mapLayer,
-        legendType: LegendItemType.MapLayer,
-        notSelectable,
-        // Overwrite fields from layer with legend fields where applicable.
-        imageRule: legendItem.imageRule ?? mapLayer.imageRule,
-        title: legendItem.title ?? mapLayer.title,
-        id: composeId(collectionId, legendItem.id),
-        minZoom: mapLayer.minZoom ?? DEFAULT_MIN_ZOOM,
-        maxZoom: DEFAULT_MAX_ZOOM,
-        noDetail: !mapLayer.detailUrl,
-        params,
-      }
-    }
-
-    // Otherwise use the plain legend item.
-    const params = legendItem.params
-      ? querystring.stringify((legendItem.params as unknown) as ParsedUrlQueryInput)
+  // Return a MapLayer if an ID is specified
+  if (legendItem.id && mapLayer) {
+    const params = mapLayer.params
+      ? querystring.stringify(mapLayer.params as ParsedUrlQueryInput)
       : undefined
 
     return {
-      ...legendItem,
-      legendType: LegendItemType.Standalone,
-      notSelectable,
+      // @ts-ignore resolveType is used by GraphQL only to handle inline fragments
+      resolveType: 'MapLayer',
```

Comment:
Instead of using `resolveType` you can use `__typename` instead, it will give you complete type information hints for the shape of `MapLayer` and `LegendItem`.

It could also be that when you provide this field the resolver should automatically be able to resolve the correct type for you, but I am not 100% sure of that so give it a test.

Reply:
Not really, the field `resolveType` is only used to set the `__resolveType` on the resolver, so GraphQL will know which type will be returned. `resolveType` can also be named something different like `resolverType`, `typename` etc.